### PR TITLE
Add support for IPP for booking orders requiring confirmation

### LINKED ISSUE #5978: Add support for accepting IPP for bookings that require confirmation
The app currently doesn't display the "Collect payment" button on booking orders with pending or approved confirmation.

Booking products + orders can be created if [the WooCommerce Bookings extension](https://woocommerce.com/products/woocommerce-bookings/?utm_source=google&utm_medium=cpc&utm_campaign=marketplace_search_brand_row&utm_content=woocommerce_+bookings&gclid=CjwKCAiAyPyQBhB6EiwAFUuakhfkEiGuTg_ns8WClwbF8EXrALCThFkLEY-WK9mw9cakoDCpDEdwiBoCl3IQAvD_BwE) is installed on the store.

Booking orders require confirmation if the store manager checks the following checkbox on the Booking product:
image

It seems that `paymentMethod` is set to `wc-booking-gateway` on such orders and this payment method is not whitelisted for IPP in the app.

### PR DESCRIPTION
**Pull request by malinajirka**
Closes: #5978 

### Description

This PR adds support for accepting IPP on booking orders that require confirmation. Such orders have `paymentMethod` set to `wc-booking-gateway` and this PR whitelists this paymentMethod for IPP.

Video with walkthrough can be found on p2-pdfdoF-x7

### Testing instructions

1. Install [WooCommerce Bookings](https://woocommerce.com/products/woocommerce-bookings/?utm_source=google&utm_medium=cpc&utm_campaign=marketplace_search_brand_row&utm_content=woocommerce_+bookings&gclid=CjwKCAiAyPyQBhB6EiwAFUuakmqSfoqQhXllPTswn-b_XK6M1kJG51Er64uHd0LYC4vpGDQClhpXERoC23wQAvD_BwE) extension on your store 
2. Create a product in WPAdmin
3. Set it's ProductType to "Bookable Product"
4. Check `Requires confirmation?` checkbox
5. Create an order and add this product to the cart
6. Proceed to checkout and select "Cash/Pay on delivery"
7. Open the order in the app
8. Collect IPP for this order

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentCollectibilityCheckerTest.kt [hunk 1]
```diff
@@ -288,6 +288,32 @@ class CardReaderPaymentCollectibilityCheckerTest : BaseUnitTest() {
...
+
+            // WHEN
+            val isCollectable = checker.isCollectable(order)
+
+            // THEN
+            assertThat(isCollectable).isTrue()
+        }
+
+    @Test
+    fun `given bookings order requires confirmation, when checking collectibility, then it is collectable`() =
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentCollectibilityCheckerTest.kt:308]: ❓ This and the above test are the same. Was there any reason behind duplicating the test with just a name change?

**Review by AnirudhBhat — APPROVED** [review]
@malinajirka Thanks for working on this. I've tested the change and it works as expected. The code change looks good as well. I've left one minor query.

**Comment by AnirudhBhat** [comment]
@malinajirka Hang on!  I'm seeing the `paymentMethod` set to `cod` for Bookable products with request confirmation checked. 

The current `trunk` branch without any change processes the IPP payment for Bookable products with request confirmation checked **successfully** since the `paymentMethod` is set to `cod`

I'm not sure if something is wrong with my product setup.

Here is my Bookable product: https://basictestsite.mystagingwebsite.com/product/bookable-price/

Screenshot 2022-03-03 at 5 36 16 PM

In short, I can successfully collect IPP for both bookable products and bookable products requiring confirmation on the latest `trunk` branch.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentCollectibilityCheckerTest.kt (hunk 1 above)
Comment (malinajirka) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentCollectibilityCheckerTest.kt:308]: It's for documentation purposes. Would you prefer if I removed one of them?

Comment (AnirudhBhat) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentCollectibilityCheckerTest.kt:308]: I see. My only concern was that if one of the test fails, the other one fails as well. The same for vice versa, if one gets fixed, the other one gets fixed as well. The person debugging might stop for a second and wonder what just happened 😄 

How about we add a comment for this saying it's for documentation purposes and there shouldn't be any surprises when this test case passes/fails when the other one passes/fails. Does this make sense?

I'm completely okay to keep it as it is as well. This is definitely a nitpick and not a blocker at all. Please go ahead and merge this PR if you think adding comments doesn't add much value.

Comment (malinajirka) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/cardreader/CardReaderPaymentCollectibilityCheckerTest.kt:308]: Makes sense, added comment in 3cbd23b41b177787184c030a3b41aa4a34403685. Thanks!
